File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260917-054908-40691/workdir/src/main/java/customerimporter/Customer.java
package customerimporter;

import java.util.ArrayList;
import java.util.List;
import java.util.function.Supplier;

public class Customer {

    private int id;
    private String firstName;
    private String lastName;
    private Identification identification;
    private List<Address> addresses;

    // initialization

    public Customer() {
        super();
        addresses = new ArrayList<>();
    }

    // addresses

    public void addAddress(Address anAddress) {
        addresses.add(anAddress);
    }

    public Address addressAt(String aStreetName, Supplier<Address> aNoneBlock) {
        for (Address address : addresses) {
            if (address.isAt(aStreetName)) return address;
        }
        return aNoneBlock.get();
    }

    public List<Address> addresses() {
        return addresses;
    }

    public boolean isAddressesEmpty() {
        return addresses.isEmpty();
    }

    // name

    public String firstName() {
        return firstName;
    }

    public void firstName(String aName) {
        firstName = aName;
    }

    public String lastName() {
        return lastName;
    }

    public void lastName(String aLastName) {
        lastName = aLastName;
    }

    // identification

    public Object cuitNumberIfNone(Supplier<Object> aNoneBlock) {
        return identification.cuitNumberIfNone(aNoneBlock);
    }

    public Object dniNumberIfNone(Supplier<Object> aNoneBlock) {
        return identification.dniNumberIfNone(aNoneBlock);
    }

    public boolean hasCUITAsIdentification() {
        return identification.isCUIT();
    }

    public boolean hasDNIAsIdentification() {
        return identification.isDNI();
    }

    public boolean isIdentifiedAs(String anIdType, String anIdNumber) {
        return identification.matches(anIdType, anIdNumber);
    }

    public String identificationNumber() {
        return identification.number();
    }

    public String identificationType() {
        return identification.type();
    }

    public void identification(Identification anIdentification) {
        identification = anIdentification;
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-211617-6449/output/project/src/customerimporter/customer.py
class Customer:

    _id = None
    _first_name = None
    _last_name = None
    _identification = None
    _addresses = None

    # initialization

    def __init__(self):
        super().__init__()
        self._addresses = []

    # addresses

    def add_address(self, an_address):
        self._addresses.append(an_address)

    def address_at(self, a_street_name, a_none_block):
        for address in self._addresses:
            if address.is_at(a_street_name): return address
        return a_none_block()

    def addresses(self):
        return self._addresses

    def is_addresses_empty(self):
        return len(self._addresses) == 0

    # name

    def first_name(self):
        return self._first_name

    def set_first_name(self, a_name):
        self._first_name = a_name

    def last_name(self):
        return self._last_name

    def set_last_name(self, a_last_name):
        self._last_name = a_last_name

    # identification

    def cuit_number_if_none(self, a_none_closure):
        return self._identification.cuit_number_if_none(a_none_closure)

    def dni_number_if_none(self, a_none_block):
        return self._identification.dni_number_if_none(a_none_block)

    def has_cuit_as_identification(self):
        return self._identification.is_cuit()

    def has_dni_as_identification(self):
        return self._identification.is_dni()

    def identification(self):
        return self._identification

    def set_identification(self, an_identification):
        self._identification = an_identification

    def identification_number(self):
        return self._identification.number()

    def identification_type(self):
        return self._identification.type_code()
